torchmetrics/metric.py

```diff
@@ -240,6 +237,33 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
         return self._forward_cache
 
+    def _reduce_state(self, state_to_reduce: Dict[str, Any]) -> None:
+        for attr in self._defaults.keys():
+            current_state = getattr(self, attr)
+            incoming_state = state_to_reduce[attr]
+            reduce_fn = self._reductions[attr]
+            if reduce_fn == dim_zero_sum:
```

Comment:
this means, that in my forward I can no longer do stuff like 

```python
self.sliding_window.append(value.detach())
if len(self.sliding_window) > self.sliding_window_size:
    self.sliding_window.pop(0)
```

As the sliding window size will only be correct after the reduction and not within the actual update, right? At least when using `forward`. This may make metrics harder to implement. In fact, I am not sure if all metrics are valid in their forward now as we mostly only test `update` and `compute`.